Foundation/NSAffineTransform.swift

```diff
@@ -38,67 +51,246 @@ public class NSAffineTransform : NSObject, NSCopying, NSSecureCoding {
+        transformStruct = transform.transformStruct.concat(transformStruct)
+    }
     
     // Transforming points and sizes
     public func transformPoint(aPoint: NSPoint) -> NSPoint {
-        let matrix = transformStruct.matrix3x3
-        let vector = Vector3(aPoint.x, aPoint.y, CGFloat(1.0))
-        let resultVector = multiplyMatrix3x3(matrix, byVector3: vector)
-        return NSMakePoint(resultVector.m1, resultVector.m2)
+        return transformStruct.applied(toPoint: aPoint)
     }
 
     public func transformSize(aSize: NSSize) -> NSSize {
-        let matrix = transformStruct.matrix3x3
-        let vector = Vector3(aSize.width, aSize.height, CGFloat(1.0))
-        let resultVector = multiplyMatrix3x3(matrix, byVector3: vector)
-        return NSMakeSize(resultVector.m1, resultVector.m2)
+        return transformStruct.applied(toSize: aSize)
     }
 
     // Transform Struct
     public var transformStruct: NSAffineTransformStruct
 }
 
-// Private helper functions and structures for linear algebra operations.
-private typealias Vector3 = (m1: CGFloat, m2: CGFloat, m3: CGFloat)
-private typealias Matrix3x3 =
-    (m11: CGFloat, m12: CGFloat, m13: CGFloat,
-     m21: CGFloat, m22: CGFloat, m23: CGFloat,
-     m31: CGFloat, m32: CGFloat, m33: CGFloat)
 
-private func multiplyMatrix3x3(matrix: Matrix3x3, byVector3 vector: Vector3) -> Vector3 {
-    let x = matrix.m11 * vector.m1 + matrix.m12 * vector.m2 + matrix.m13 * vector.m3
-    let y = matrix.m21 * vector.m1 + matrix.m22 * vector.m2 + matrix.m23 * vector.m3
-    let z = matrix.m31 * vector.m1 + matrix.m32 * vector.m2 + matrix.m33 * vector.m3
+private extension NSAffineTransformStruct {
+    /**
+     Creates an affine transformation matrix from translation values.
+     The matrix takes the following form:
+     
+         [ 1  0  tX ]
+         [ 0  1  tY ]
+         [ 0  0   1 ]
+     */
+    static func translation(tX tX: CGFloat, tY: CGFloat) -> NSAffineTransformStruct {
+        return NSAffineTransformStruct(
+            m11: CGFloat(1.0), m12: CGFloat(),
+            m21: CGFloat(),    m22: CGFloat(1.0),
+            tX: tX, tY: tY
+        )
+    }
+    
+    /**
+     Creates an affine transformation matrix from scaling values.
+     The matrix takes the following form:
+     
+         [ sX   0  0 ]
+         [ 0   sY  0 ]
+         [ 0    0  1 ]
+     */
+    static func scale(sX sX: CGFloat, sY: CGFloat) -> NSAffineTransformStruct {
+        return NSAffineTransformStruct(
+            m11: sX, m12: CGFloat(),
+            m21: CGFloat(), m22: sY,
+            tX: CGFloat(), tY: CGFloat()
+        )
+    }
+    
+    /**
+     Creates an affine transformation matrix from rotation value (angle in radians).
+     The matrix takes the following form:
+     
+         [ cos α   -sin α  0 ]
+         [ sin α    cos α  0 ]
+         [   0        0    1 ]
+     */
+    static func rotation(radians angle: CGFloat) -> NSAffineTransformStruct {
+        let α = Double(angle)
```

Comment:
I think it makes sense to use it in limited places like this, where it actually conveys some meaning.